### PR TITLE
Expand further on <> syntax in TRPL.

### LINKED ISSUE #26917: Explain the `<>` more, in or before the Lifetimes chapter of TRPL where it first appears
Some [feedback from reddit](https://www.reddit.com/r/rust/comments/3cl12r/is_rust_too_complicated/csx0y6t):

> Also, the function<blah>syntax hadn't come up previously [to the Lifetimes chapter], so it would be nice to expand on that.

The Lifetimes chapter currently has:

> Before we get to that, though, let’s break the explicit example down:
> 
>     fn bar<'a>(...)
>
> This part declares our lifetimes. This says that bar has one lifetime, 'a. If we had two reference parameters, it would look like this:
>
>     fn bar<'a, 'b>(...)
>
> Then in our parameter list, we use the lifetimes we’ve named:

which does say "declares our lifetimes", so maybe further explanation isn't appropriate here, but it looks like the Functions chapter is sticking to the bare minimum. 

I'm not sure where this would fit, but I think the `< >` could use another sentence or two of explanation-- "This is where you declare lifetime and generic type parameters that this function is going to use" and link to the appropriate chapters for that perhaps?

### PR DESCRIPTION
**Pull request by steveklabnik**
We haven't discussed this syntax yet, so provide a basic explanation
and link up to later chapters.

Fixes #26917

### CODE REVIEW

**Comment by Gankro** [comment]
No turbo fish? :crying_cat_face:

File: src/doc/trpl/lifetimes.md [hunk 1]
```diff
@@ -77,8 +77,18 @@ Before we get to that, though, let’s break the explicit example down:
...
+discuss the `<>`s after a function’s name. A function can have ‘generic
+parameters’ between the `<>`s, of which lifetimes are one kind. We’ll discuss
+other kinds of generics [later in the book][generics], but for now, let’s
+just focus on the lifteimes aspect.
+
+[functions]: functions.html
+[generics]: generics.html
+
+We use `<>` to declare our lifetimes. This says that `bar` has one lifetime,
+`'a`. If we had two reference parameters, it would look like this:
```
Comment (Gankro) [inline src/doc/trpl/lifetimes.md]: This might be a bit deceptive -- a function can have many more lifetimes that aren't declared thanks to elision.

Comment (alexcrichton) [inline src/doc/trpl/lifetimes.md]: From a technical perspective I each function can have multiple lifetime parameters, but from the perspective of just explaining syntax I think it's fine to assume that "lifetime parameter" is the same as "explicitly declared lifetime parameter". Elision kinda fits into the same system later on, so I don't think that we're going to confuse anyone from this.

**Comment by alexcrichton** [comment]
@bors: r+ 8f828a3a9d66a3ff2465861f1d8f12dd0f088d23 rollup

(but feel free to r- @Gankro if you disagree)

**Comment by Gankro** [comment]
*shrug* monotonic improvement -- can always refine later if desired.

**Pull request merged**